### PR TITLE
test: Extend rdmsr test for new Intel templates

### PR DESCRIPTION
**Pull request by sudanl0**
Signed-off-by: Sudan Landge <sudanl@amazon.com>

## Changes

- Capture baseline files for T2CL template by running msr_reader.sh on host/guest running kernel 4.14/5.10 with the bionic-msrtools rootfs on m6i.metal.
- The implemented state of MSRs on Skylake conflict with the state of MSRs on Cascadelake so create a template exception list to avoid running T2CL on Skylake.

## Reason

We need to support T2CL template in rdmsr test.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

~~- [ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
~~- [ ] API changes follow the [Runbook for Firecracker API changes][2].~~
~~- [ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
~~- [ ] New `TODO`s link to an issue.~~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

~~- [ ] This functionality can be added in [`rust-vmm`][1].~~

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

File: tests/integration_tests/functional/test_cpu_features.py [hunk 1]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
 
-# CPU templates which support MSR test
-MSR_SUPPORTED_TEMPLATES = ["T2A", "T2S"]
+
+def get_msr_supported_templates():
+    """
+    Return the list of CPU templates supported for MSR-related tests.
+    """
+    # CPU templates supported for the MSR tests
+    _msr_supported_templates = ["T2A", "T2S"]
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:205]: nit: any reason it starts from underscore?

File: tests/integration_tests/functional/test_cpu_features.py [hunk 2]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+
+    # CPU Templates which need additional checks are added below:
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:207]: nit: s/Templates/templates/

File: tests/integration_tests/functional/test_cpu_features.py [hunk 3]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+
+    # cascadelake on m5d.metal has MSR 0x122 state as implemented whereas,
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:209]: nit: s/cascadelake/Cascade Lake/

File: tests/integration_tests/functional/test_cpu_features.py [hunk 4]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+    # skylake on m5d.metal has MSR 0x122 state as unimplemented.
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:210]: nit: s/skylake/Skylake/

File: tests/integration_tests/functional/test_cpu_features.py [hunk 5]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+    # Since the conflict is seen only with skylake and cascade lake,
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:211]: nit: same as above

File: tests/integration_tests/functional/test_cpu_features.py [hunk 6]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+    # we add T2CL(cascadelake) template only when CPU is not skylake.
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:212]: nit: missing space

File: tests/integration_tests/functional/test_cpu_features.py [hunk 7]
```diff
@@ -196,8 +196,31 @@ def test_brand_string(test_microvm_with_api, network_config):
...
+    t2cl_exception_list = [
+        # Note: we may need to update this list if there are
+        # more conflicting CPU models reported in the future.
+        "Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz",  # skylake
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_cpu_features.py:216]: nit: same as above
Also, unify the names in the commit message.

**Review by zulinx86 — COMMENTED** [review]
Looks good to me other than what Nikita has already commented.

File: tests/integration_tests/functional/test_cpu_features.py (hunk 1 above)
Comment (sudanl0) [reply tests/integration_tests/functional/test_cpu_features.py:205]: underscore was added to highlight that this is a private variable. If it isn't inline with our coding conventions I can remove it.

Comment (kalyazin) [reply tests/integration_tests/functional/test_cpu_features.py:205]: No, I'm not aware of such a convention. At least by looking at this file I can't see other examples like that.
The only similar thing that come to my mind is __test_*() functions that are prepended with an underscore to hide them from pytest.

**Review by zulinx86 — APPROVED** [review]

**Review by kalyazin — APPROVED** [review]

**Pull request merged**